Orm/Xtensive.Orm/Orm/Internals/CompiledQueryRunner.cs

```diff
@@ -192,6 +199,25 @@ private void AllocateParameterAndReplacer()
         }
         return null;
       });
+
+      return !closureType.Name.Contains("<>c__DisplayClass")            // 'DisplayClass' is generated class for captured objects
```

Comment:
Take a look at ```TypeHelper.IsClosure()``` extension method. Is it what you wanted to achieve here?

Reply:
Yes